src/GovUk.Education.ExploreEducationStatistics.Common/Extensions/ReflectionExtensions.cs

```diff
@@ -58,12 +58,7 @@ public static bool TryBoxToResult(this object value, Type targetType, out object
                         break;
 
                     case "Task":
-                        var task = typeof(Task).GetMethod("FromResult")
-                            ?.MakeGenericMethod(boxType.GenericTypeArguments)
-                            .Invoke(null, new[] { boxedValue });
-
-                        boxedValue = task;
-                        break;
+                        throw new ArgumentException("Cannot unbox Tasks as this may cause thread exhaustion. Consider awaiting the result first.");
```

Comment:
Repeat of another comment - would reinstate the previous behaviour as this actually unsafe as we're not calling any synchronously blocking methods like `Result`. All test relevant test cases should also be reinstated.

Also this exception doesn't even make sense as we're not unboxing the result.